app/assets/javascripts/discourse/models/post.js.es6

```diff
@@ -320,6 +320,16 @@ Post.reopenClass({
     });
   },
 
+  mergePosts(selectedPosts, selectedReplies) {
+    return Discourse.ajax("/posts/merge_posts", {
+      type: 'PUT',
+      data: {
+        post_ids: selectedPosts.map(function(p) { return p.get('id'); }),
+        reply_post_ids: selectedReplies.map(function(p) { return p.get('id'); })
+      }
+    });
```

Comment:
It seems like we're not updating the client and handling errors on the ajax calls here.